docs/integrations/url-reader.md

```diff
@@ -0,0 +1,227 @@
+---
+id: url-reader
+title: URL Reader
+sidebar_label: URL Reader
+# prettier-ignore
+description: URL Reader is a backend core API responsible for reading files from external locations.
+---
+
+## Concept
+
+Some of the core plugins of Backstage have to read files from an external
+location. [Software Catalog](../features/software-catalog/index.md) has to read
+the [`catalog-info.yaml`](../features/software-catalog/descriptor-format.md)
+entity descriptor files to register and track an entity.
+[Software Templates](../features/software-templates/index.md) have to download
+the template skeleton files before creating a new component.
+[TechDocs](../features/techdocs/README.md) has to download the markdown source
+files before generating a documentation site.
+
```

Comment:
Aah good point. I'd go with URL Readers belong in @backstage/backend-common package and Catalog/TechDocs/<any other plugin> use them for their purpose.

(This should go into the TechDocs technical documentation) but TechDocs has two use cases of "reading" files. The first one is known as the ["prepare" step](https://backstage.io/docs/features/techdocs/concepts#techdocs-preparer) which is to download all the markdown files so that a documentation site can be generated.

The second "reading" use case is when a user is navigating TechDocs sites and an external storage system (e.g. S3) is configured. This part does not use existing URL Readers, and has its own implementation. They are also called ["TechDocs publishers"](https://backstage.io/docs/features/techdocs/concepts#techdocs-publisher) since they are also used for publishing generated HTML and other static files to the bucket.